src/schema/field_entry.rs

```diff
@@ -168,8 +181,16 @@ impl<'de> Deserialize<'de> for FieldEntry {
                                 return Err(de::Error::duplicate_field("type"));
                             }
                             ty = Some(map.next_value()?);
-                            if ty == Some("hierarchical_facet") {
-                                field_type = Some(FieldType::HierarchicalFacet);
+                            match ty {
```

Comment:
something is weird in this code (not yours). We wrap it in `Some` and match it right away?

Reply:
Can you have a look at the bigger picture and see if there is something wrong here?

Reply:
Yeah, will do.